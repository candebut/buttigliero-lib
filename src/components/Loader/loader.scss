.loaderWrapper {
  width: 100%;
  height: 100%;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
}

.loaderComponent {
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
  display: block;
  height: 10px;
  width: 10px;
  -webkit-animation: spin 2s linear infinite;
  animation: spin 2s linear infinite;
}

@-webkit-keyframes spin {
  from {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    -webkit-transform: rotate(359deg);
    transform: rotate(359deg);
  }
}

@keyframes spin {
  from {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    -webkit-transform: rotate(359deg);
    transform: rotate(359deg);
    -webkit-transform: rotate(359deg);
    transform: rotate(359deg);
  }
}
