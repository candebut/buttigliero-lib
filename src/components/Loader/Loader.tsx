import './loader.scss';
import React from 'react';

type Props = {
  children: string | JSX.Element | JSX.Element[];
};

function Loader({ children }: Props) {
  return (
    <div className="loaderWrapper">
      <div className="loaderComponent">{children}</div>
    </div>
  );
}

export default Loader;
