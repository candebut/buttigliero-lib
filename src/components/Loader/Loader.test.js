import React from 'react';
import Loader from './Loader';
import { shallow, mount } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('Loader: snapshot tests', () => {
  it('should render correctly in primary variant', () => {
    const component = shallow(<Loader>Loading test</Loader>);
    expect(component).toMatchSnapshot();
  });
});
