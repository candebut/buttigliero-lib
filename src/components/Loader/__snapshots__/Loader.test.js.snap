// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader: snapshot tests should render correctly in primary variant 1`] = `
<div
  className="loaderWrapper"
>
  <div
    className="loaderComponent"
  >
    Loading test
  </div>
</div>
`;
